examples/c/profile.c

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+/* Copyright (c) 2022 Facebook */
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include <sys/sysinfo.h>
+#include <linux/perf_event.h>
+#include <bpf/libbpf.h>
+#include <bpf/bpf.h>
+
+#include "profile.skel.h"
+#include "profile.h"
+#include "blazesym.h"
+
+
+static long
+perf_event_open(struct perf_event_attr *hw_event, pid_t pid,
+		int cpu, int group_fd, unsigned long flags)
+{
+	int ret;
+
+	ret = syscall(__NR_perf_event_open, hw_event, pid, cpu,
+		      group_fd, flags);
+	return ret;
+}
+
+struct handler_ctx {
```

Comment:
for a simple tool like this it is ok to use static global variables to keep this process-wide state. We do that a lot in libbpf-tools and it works totally fine. The only downside to that is worse testability through unit tests, but no one is going to write unit tests for these demos anyways, so there is no need to add extra verboseness for this. Keep it simple.